### PR TITLE
Homework21: Alina Shautsova

### PR DESCRIPTION
**Pull request by AlinaShautsova**

### CODE REVIEW

**Comment by AlinaShautsova** [comment]
HTML files are located: 
pymar2024\tests\report_bank.html           and 
pymar2024\tests\report_library.html

**Comment by Stml89** [comment]
First of all, remove `.mypy_cache` directory

File: homework21/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+from homework21.source.homework11_bank_deposit import Bank, Deposit
+from homework21.source.homework11_library import Book, User
+from logging_module import setup_logging
+
+
+@pytest.fixture(scope="session", autouse=True)
+def logger():
+    """Set up logger."""
+    logger = setup_logging()
+    return logger
```
Comment (Stml89) [inline homework21/tests/conftest.py]: you can just import it to module and use it

File: homework21/tests/test_homework21_bank.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""Pytest for Homework 11: bank deposit."""
+import pytest
+
+
+def test_deposit_initialization(setup_deposit, logger):
```
Comment (Stml89) [inline homework21/tests/test_homework21_bank.py]: just import logger and use it here

File: homework21/tests/test_homework21_bank.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    logger.info("Testing that the initial amount is less than the final "
+                "amount.")
+    final_amount = setup_deposit.final_amount()
+    logger.debug("Initial amount: %f, final amount: %f",
+                 setup_initial_amount, final_amount)
+    assert setup_initial_amount < final_amount
+
+
+if __name__ == '__main__':
+    pytest.main()
```
Comment (Stml89) [inline homework21/tests/test_homework21_bank.py]: you do not need it here, pytest can works without it

File: homework21/tests/test_homework21_library.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    setup_user1_library.take_book(setup_book)
+    assert setup_book.took_by == setup_user1_library.name
+    setup_user2_library.return_book(setup_book)
+    logger.debug("Book was not returned_by after it was taken by another "
+                 "user: %s", setup_book.took_by)
+    assert setup_book.took_by == setup_user1_library.name
+
+
+if __name__ == "__main__":
+    pytest.main()
```
Comment (Stml89) [inline homework21/tests/test_homework21_library.py]: the same

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework21/tests/test_homework21_library.py (hunk 4 above)
Comment (AlinaShautsova) [reply homework21/tests/test_homework21_library.py]: Thank you, I fixed everything

File: homework21/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+from homework21.source.homework11_bank_deposit import Bank, Deposit
+from homework21.source.homework11_library import Book, User
+from logging_module import setup_logging
+
+
+logger = setup_logging()
+
+
+@pytest.fixture(scope="function")
```
Comment (Stml89) [inline homework21/tests/conftest.py]: function scope by default , so you do not need to write it everytime
```suggestion
@pytest.fixture()
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework21/tests/conftest.py (hunk 5 above)
Comment (AlinaShautsova) [reply homework21/tests/conftest.py]: Done!

**Review by Stml89 — APPROVED** [review]

**Pull request merged**